packages/node_modules/nav-frontend-toggle/src/toggle-knapp-pure.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import * as PT from 'prop-types';
+import { HTMLAttributes } from 'react';
+import * as cn from 'classnames';
+
+import { omit, guid } from 'nav-frontend-js-utils';
+import 'nav-frontend-toggle-style';
+
+const knappCls = (props) => cn('knapp', 'toggleKnapp', {
+    'toggleKnapp--pressed': props.pressed,
+    'toggleKnapp--kompakt': props.kompakt
+});
+
+export interface ToggleKnappPureProps extends HTMLAttributes<HTMLButtonElement> {
+    /**
+     * Pressed state
+     */
+    pressed?: boolean;
+    /**
+     * Kompakt versjon
+     */
+    kompakt?: boolean;
+}
+
+class ToggleKnappPure extends React.PureComponent<ToggleKnappPureProps> {
+    static defaultProps: ToggleKnappPureProps = {
+        pressed: false
+    };
+    render() {
+        const renderProps = omit(this.props, 'children', 'pressed', 'kompakt');
+        const knappId = guid();
+        return (
+            <button
```

Comment:
Er skikkelig på tynn is her, men hvordan er dette UU-messig? Hvis jeg har forstått det rett som at disse er tenkt som skjemaelementer, hvordan er det å bruke `button` til det?

Reply:
Det er primært UU-hensyn som gjør at toggle-komponenten skrives om fra å være radio-inputs til button-elementer (som med `aria-pressed` attributtet blir opplest som "toggle button" av skjermlesere). Den nye versjonen er ikke tenkt å være et rent skjema-element, men skal kunne brukes fritt hvor som helst - også inne i forms. Sålenge knappen har attributtet `type="button"` vil ikke det være noe problem ([ref](https://dev.to/claireparker/why-its-important-to-give-your-html-button-a-type-58k9)).